src/resmom/catch_child.c

```diff
@@ -2735,18 +2735,30 @@ send_sisters_deljob_wait(job *pjob)
  * 	rid_job - rid mom of a job that the server says no longer exists
  *
  * @param[in] jobid - char pointer holding jobid
+ * @param[in] msg - message to log if job has been deleted.
  *
  * @return Void
  *
  */
 void
-rid_job(char *jobid)
+rid_job(char *jobid, char *msg)
 {
 	job	*pjob;
 
 	pjob = find_job(jobid);
-	if (pjob && !pjob->ji_hook_running_bg_on)
+	if (pjob && !pjob->ji_hook_running_bg_on && (pjob->ji_qs.ji_substate != JOB_SUBSTATE_PRERUN)) {
```

Comment:
Just so I am clear... if the job has started, it already sent an update to the server so there is no longer a race condition in that case, correct? If so, you might want to mention that in the comment.

Reply:
Correct. I'll add to the comment.